### PR TITLE
[Core] Raise minimum PHP requirement to 7.3

### LINKED ISSUE #5063: LORIS should start requiring PHP 7.3 on minor + bugfix (and 7.4 on major when it's out)
PHP 7.4 comes out at the end of November 2019. At the same time, PHP 7.2 will no longer have active support. 

https://www.php.net/supported-versions.php

On the major branch we should start using PHP 7.3 (and discontinue 7.2 testing) as well as enable PHP 7.4 testing on all branches once it's out.

Importantly, 7.4 will allow us to use strong types in PHP class variables.

Requiring all LORIS instances to use >= 7.3 also means they'll have guaranteed CSRF protection (#4083)

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

PHP 7.2 is no longer receiving active support as of Nov. 30th, 2019. PHP 7.4 is being released at about the same time.

Typically we only support the last two releases of PHP. This PR brings the minimum requirement up to 7.3 from 7.2.

Detailed changes between these versions can be found here: https://www.php.net/manual/en/migration73.php

I'm adding the Security label because this upgrade will enable all LORIS instances to be fully protected from CSRF attacks via the SameSite cookie.

#### Links to related tickets (GitHub, Redmine, ...)

* Resolves #5063

### CODE REVIEW

**Comment by PapillonMcGill** [comment]
Will also require to update README.CentOS7.md files (currently in 22.0-release).

**Comment by johnsaigle** [comment]
PHP 7.3 is required to use phpstan, so this is blocking #5356.

**Comment by johnsaigle** [comment]
PHP 7.4 is out so this PR can go ahead.

It's possible that Travis doesn't yet support this in the build environments though.

https://www.php.net/releases/7_4_0.php

**Comment by johnsaigle** [comment]
Adding Blocked because TravisCI doesn't yet support testing against the PHP 7.4 release.

**Comment by johnsaigle** [comment]
Blocked by issue #5651

**Comment by driusan** [comment]
This is no longer blocked, phan was updated.

**Comment by johnsaigle** [comment]
@driusan Please review as this is no longer blocked by phan and is also holding up other PRs.

File: modules/create_timepoint/php/create_timepoint.class.inc [hunk 1]
```diff
@@ -85,13 +85,13 @@ class Create_Timepoint extends \NDB_Form
...
             //if there is only one project, autoselect first project from array of 1
-            //TODO: change this to array_key_first() when support is only PHP 7.3+
-            $project = \Project::getProjectFromID(array_pop($user_list_of_projects));
+            $project_id = $user_list_of_projects[0];
         } else if (count($user_list_of_projects) > 1) {
             $project_id = intval($values['project']);
-            $project    = \Project::getProjectFromID($project_id);
         }
 
+        $project = \Project::getProjectFromID($project_id);
```
Comment (driusan) [inline modules/create_timepoint/php/create_timepoint.class.inc]: This looks like a change in behaviour? Why is this required for 7.3?

Comment (driusan) [reply modules/create_timepoint/php/create_timepoint.class.inc]: other than that if it's passing Travis, it looks like changes are minimal so it's good enough for me.

Comment (johnsaigle) [reply modules/create_timepoint/php/create_timepoint.class.inc]: It's not required. It was an enhancement but I agree it's not very helpful here. I'll revert this file.
